Index: git/handlers/raw_handler.c
===================================================================
--- git.orig/handlers/raw_handler.c
+++ git/handlers/raw_handler.c
@@ -85,14 +85,16 @@ static int install_raw_file(struct img_t
 		}
 	}
 
-	TRACE("Installing file %s on %s\n",
-		img->fname, path);
-	fdout = openfileoutput(path);
-	ret = copyimage(&fdout, img, NULL);
-	if (ret< 0) {
-		ERROR("Error copying extracted file\n");
+	if (strcmp(img->path, path)) {
+		TRACE("Installing file %s on %s\n",
+			img->fname, path);
+		fdout = openfileoutput(path);
+		ret = copyimage(&fdout, img, NULL);
+		if (ret< 0) {
+			ERROR("Error copying extracted file\n");
+		}
+		close(fdout);
 	}
-	close(fdout);
 
 	if (use_mount) {
 		umount(DATADST_DIR);
